backend/asmgen.ml

```diff
@@ -167,35 +165,19 @@ let should_use_linscan fd =
   List.mem Cmm.Use_linscan_regalloc fd.Mach.fun_codegen_options
 
 let if_emit_do f x = if should_emit () then f x else ()
```

Comment:
(I wonder whether the code would be
slightly easier to understand by
eta-expanding the uses if `if_emit_do`;
but non-expanded uses were present
before this pull request.)